Components/Core/Inc/Timer.hpp

```diff
@@ -25,21 +35,34 @@ constexpr uint32_t DEFAULT_TIMER_COMMAND_WAIT_PERIOD = MS_TO_TICKS(15); // Defau
 class Timer
 {
 public:
-	Timer();
-
-	bool ChangePeriod(const uint32_t period);
-
-	// WORK-IN-PROGRESS
-	// NOTES:
-	// - I can think of several timer types
-	// 1) Default Ctor Timer (1 second polling timer that requires polling to acquire state with no callback)
-	// 2) Callback Enabled Timer (user-provided callback)
+	Timer(); // Default Constructor (Polling Timer)
+	Timer(void (*TimerCallbackFunction_t)( TimerHandle_t xTimer )); // Constructor for Callback Enabled Timer
+	~Timer();
+	bool ChangePeriodMs(const uint32_t period_ms);
+	bool ChangePeriodMsAndStart(const uint32_t period_ms);
+	bool Start();
+	bool Stop();
+	bool ResetTimer();
+	bool ResetTimerAndStart();
+	void SetAutoReload(bool setReloadOn);  //True for autoreload and False for One-shot
 
+	const bool GetIfAutoReload();
+	const TimerState GetState();
+	const uint32_t GetPeriodMs();
+	const uint32_t GetRemainingTimeMs();
 
+	static void DefaultCallback( TimerHandle_t xTimer );
 
 protected:
-	TimerHandle_t rtTimerHandle;
+	const uint32_t GetRTOSTimeRemaining();
+
+	TimerState timerState; // Enum that holds current timer state
+	TimerHandle_t rtTimerHandle; //what
+	uint32_t remainingTime;
```

Comment:
Not needed, need to worry about the original timer period after a start/stop/complete.

You will need to restore the `originalTimerPeriod` in these cases:
- On `ResetTimer()` / `ResetTimerAndStart()`
- Whenever the timer is complete (inside the `TimerCallback`)